wire-ios-sync-engine/Source/Use cases/SnoozeCertificateEnrollmentUseCase.swift

```diff
@@ -72,10 +74,14 @@ final class SnoozeCertificateEnrollmentUseCase: SnoozeCertificateEnrollmentUseCa
             id: actionId,
             interval: interval
         ) {
-            // We save the end of the grace period once and should not update it.
-            let notificationObject = FeatureRepository.FeatureChange.e2eIEnabled(gracePeriod: nil)
-            NotificationCenter.default.post(name: .featureDidChangeNotification,
-                                            object: notificationObject)
+            if isUpdateMode {
+                NotificationCenter.default.post(name: .checkForE2EICertificateExpiryStatus, object: nil)
+            } else {
+                // We save the end of the grace period once and should not update it.
+                let notificationObject = FeatureRepository.FeatureChange.e2eIEnabled(gracePeriod: nil)
+                NotificationCenter.default.post(name: .featureDidChangeNotification,
+                                                object: notificationObject)
+            }
```

Comment:
Could you please explain to me briefly, when this code might be executed?
When is `RecurringActionService.performActionsIfNeeded()` called?

Reply:
This is a reminder service. There's a calculation in `SnoozeTimeProvider` based on which the next reminder alert would be shown for enrolling or updating a certificate.